### PR TITLE
Draft: Allow GraphQL via ZMQ 

### PR DESCRIPTION
**Pull request by felixvd**
As discussed, this allows sending GraphQL queries via ZMQ.

See https://github.com/mujin/controllerclientcpp/pull/124

### CODE REVIEW

File: python/mujinwebstackclient/webstackgraphclientutils.py [hunk 1]
```diff
@@ -36,8 +36,17 @@ class GraphClientBase(object):
...
         self._webclient = webclient
+        self._webstackZmqClient = zmqClient
+
+    def _CallGraphAPIV2(self, query, variables=None, timeout=5.0):
+        payload = {
+            'path': '/api/v2/graphql',
+            'query': query,
+            'variables': variables,
+        }
+        return self._webstackZmqClient.SendCommand(payload, timeout=timeout)
```
Comment (cielavenir) [inline python/mujinwebstackclient/webstackgraphclientutils.py:49]: @felixvd according to test_planningsystem MR output, this SendCommand requires checkpreempt argument

**Review by cielavenir — CHANGES_REQUESTED** [review]
Working on test_planningsystem revealed that a change like https://github.com/mujin/mujinwebstackclientpy/pull/11/files is required.

If wrong, please show the proper webstack branch.

File: python/mujinwebstackclient/webstackgraphclientutils.py (hunk 1 above)
Comment (cielavenir) [reply python/mujinwebstackclient/webstackgraphclientutils.py]: Please add checkpreempt=True or checkpreempt=False, though it seems just giving warning to stderr...

**Review by cielavenir — DISMISSED** [review]
It works

File: python/mujinwebstackclient/webstackgraphclientutils.py [hunk 2]
```diff
@@ -36,8 +36,17 @@ class GraphClientBase(object):
...
         self._webclient = webclient
+        self._webstackZmqClient = zmqClient
+
+    def _CallGraphAPIV2(self, query, variables=None, timeout=5.0):
+        payload = {
+            'path': '/api/v2/graphql',
+            'query': query,
+            'variables': variables,
+        }
+        return self._webstackZmqClient.SendCommand(payload, timeout=timeout)['data']
```
Comment (cielavenir) [inline python/mujinwebstackclient/webstackgraphclientutils.py:49]:
```suggestion
        return self._webstackZmqClient.SendCommand(payload, checkpreempt=False, timeout=timeout)['data']
```

Note: checkpreempt=False is from simulator.mujin.supervisordclient, pretty not sure

**Comment by cielavenir** [comment]
@felixvd need `SendCommand(sendjson=False)` if payload is msgpack

**Comment by felixvd** [comment]
We ended up separating the internal client.

**Comment by cielavenir** [comment]
cf https://github.com/mujin/controllerclientcpp/pull/124#issuecomment-1506272878

**Pull request closed without merging**